packages/commons/src/config/featureFlagsConfig.ts

```diff
@@ -1,20 +1,74 @@
+import { notFound } from "pagopa-interop-models";
 import { z } from "zod";
 
-export const FeatureFlagsConfig = z
+export const FeatureFlagSignalhubConfig = z
   .object({
     FEATURE_FLAG_SIGNALHUB_WHITELIST: z
       .enum(["true", "false"])
       .default("false")
-      .transform((value) => value === "true"),
+      .transform((value) => value === "true")
+      .optional(),
     SIGNALHUB_WHITELIST_PRODUCER: z
       .string()
       .transform((value) => value.split(","))
       .pipe(z.array(z.string().uuid()))
       .optional(),
   })
   .transform((c) => ({
-    featureFlagSignalhubWhitelist: c.FEATURE_FLAG_SIGNALHUB_WHITELIST,
+    featureFlagSignalhubWhitelist: c.FEATURE_FLAG_SIGNALHUB_WHITELIST ?? false,
     signalhubWhitelistProducer: c.SIGNALHUB_WHITELIST_PRODUCER,
   }));
 
-export type FeatureFlagsConfig = z.infer<typeof FeatureFlagsConfig>;
+export type FeatureFlagSignalhubConfig = z.infer<
+  typeof FeatureFlagSignalhubConfig
+>;
+
+export const FeatureFlagAgreementApprovalPolicyUpdateConfig = z
+  .object({
+    FEATURE_FLAG_AGREEMENT_APPROVAL_POLICY_UPDATE: z
+      .enum(["true", "false"])
+      .default("false")
+      .transform((value) => value === "true")
+      .optional(),
+  })
+  .transform((c) => ({
+    featureFlagAgreementApprovalPolicyUpdate:
+      c.FEATURE_FLAG_AGREEMENT_APPROVAL_POLICY_UPDATE ?? false,
+  }));
+
+export type FeatureFlagAgreementApprovalPolicyUpdateConfig = z.infer<
+  typeof FeatureFlagAgreementApprovalPolicyUpdateConfig
+>;
+
+export const isFeatureFlagEnabled = <
+  T extends Record<string, unknown>,
+  K extends keyof T & string
+>(
+  config: T,
+  featureFlagName: K extends `featureFlag${string}` ? K : never
+): boolean => {
+  const featureFlag = config[featureFlagName];
+
+  if (featureFlag === undefined) {
+    return false;
+  }
+
+  return featureFlag === true;
+};
+export const assertFeatureFlag = <
```

Comment:
Proposal: let's add also here the "Enabled" suffix as in `isFeatureFlagEnabled`

```suggestion
export const assertFeatureFlagEnabled = <
```